adapters/Mintegral/MintegralAdapter/GADMediationAdapterMintegral.m

```diff
@@ -4,44 +4,100 @@
+#import "GADMAdapterMintegralRewardedAdLoader.h"
+#import "GADMAdapterMintegralBannerAdLoader.h"
+#import "GADMAdapterMintegralInterstitialAdLoader.h"
+#import "GADMAdapterMintegralNativeAdLoader.h"
+#import "GADMAdapterMintegralUtils.h"
 
-@implementation GADMediationAdapterMintegral
+#import <MTGSDK/MTGSDK.h>
+#import <MTGSDKBidding/MTGBiddingSDK.h>
 
-- (void)collectSignalsForRequestParameters:(nonnull GADRTBRequestParameters *)params
-                         completionHandler:
-                             (nonnull GADRTBSignalCompletionHandler)completionHandler {
-  // TODO: Start signal generation and call completionHandler when the signal generation finishes or
-  // fails.
-  completionHandler("", nil);
+
+@implementation GADMediationAdapterMintegral {
+    /// Mintegral rewarded ad.
+    GADMAdapterMintegralRewardedAdLoader * _rewardedAd;
+    
+    /// Mintegral banner ad.
+    GADMAdapterMintegralBannerAdLoader *_bannerAd;
+    
+    /// Mintegral interstitial ad.
+    GADMAdapterMintegralInterstitialAdLoader *_interstitialAd;
+    
+    /// Mintegral native ad.
+    GADMAdapterMintegralNativeAdLoader *_nativeAd;
+}
+
+
+#pragma mark GADRTBAdapter
+
++ (nullable Class<GADAdNetworkExtras>)networkExtrasClass {
+    return [GADMAdapterMintegralExtras class];
 }
 
 + (void)setUpWithConfiguration:(nonnull GADMediationServerConfiguration *)configuration
              completionHandler:(nonnull GADMediationAdapterSetUpCompletionBlock)completionHandler {
-  // TODO: Set up Mintegral SDK and perform any necessary prefetching or configuration work.
-  completionHandler(nil);
+    
+    NSMutableSet *appIds = [[NSMutableSet alloc] init];
+    NSMutableSet *appKeys = [[NSMutableSet alloc] init];
+    
+    for (GADMediationCredentials *credential in configuration.credentials) {
+        NSString *appId = credential.settings[GADMAdapterMintegralAppID];
+        GADMAdapterMintegralMutableSetAddObject(appIds, appId);
+        
+        NSString *appKey =credential.settings[GADMAdapterMintegralAppKey];
+        GADMAdapterMintegralMutableSetAddObject(appKeys, appKey);
+    }
+    
+    if (appIds.count < 1 || appKeys.count < 1) {
+        NSError *error = GADMAdapterMintegralErrorWithCodeAndDescription(GADMintegralErrorInvalidServerParameters, @"Mintegral mediation configurations did not contain a valid app ID or app Key.");
+        completionHandler(error);
+        return;
+    }
+    
+    NSString *appId = [appIds anyObject];
+    if (appIds.count > 1) {
+        GADMediationAdapterMintegralLog(
+          @"Found the following App IDs:%@. Please remove any app IDs which you are not using",
+          appIds);
+        GADMediationAdapterMintegralLog(@"Configuring Mintegral SDK with the app ID:%@", appId);
+    }
+    
+    NSString *appKey = [appKeys anyObject];
+    if (appKeys.count > 1) {
+        GADMediationAdapterMintegralLog(
+          @"Found the following Api keys:%@. Please remove any api keys which you are not using",
```

Comment:
`App Keys` and add period at the end after "using".

Reply:
OK, I've made changes and committed.